### PR TITLE
Fixing the gradient for cyclic ranges

### PR DESCRIPTION
**Pull request by vascotenner**
The gradient operation wraps the gradient when the input has a cyclic range see #238

### CODE REVIEW

File: holoviews/operation/element.py [hunk 1]
```diff
@@ -316,8 +316,12 @@ def _process(self, matrix, key=None):
         dx = np.diff(data, 1, axis=1)[0:r-1, 0:c-1]
         dy = np.diff(data, 1, axis=0)[0:r-1, 0:c-1]
 
-        cyclic_range = 1.0 if not matrix_dim.cyclic else matrix_dim.range
+        cyclic_range = None if not matrix_dim.cyclic else np.diff(matrix_dim.range)
```
Comment (jlstevens) [inline holoviews/operation/element.py]: Using ``np.diff(matrix_dim.range)`` if fine assuming that both values are numeric. This assumption may be violated as the range tuple is allowed to use a ``None`` value to indicate lower/upper bounds that are not set. For this reason, I would have an extra line with something like:

```if  matrix_dim.cyclic and (None in matrix_dim.range):
   raise Exception('Cyclic range must be specified to compute the gradient over cyclic quantities')

**Comment by jlstevens** [comment]
I've made one comment but other than that, this PR looks fine to me.

**Comment by jbednar** [comment]
Jean-Luc's suggestion sounds good.  The code looks good to me, but it sure would be nice to see test cases. :-)

**Comment by vascotenner** [comment]
I just did some test, there are some fundamental problems. I prefer that if a gradient is zero, it should stay zero, no matter what wrapping is done!

**Comment by vascotenner** [comment]
Zero gradient stays zero, and small steps are preferred above large ones.

**Comment by jlstevens** [comment]
Do we think this is now ready to merge?

**Comment by vascotenner** [comment]
This PR makes that the gradient operator works again, and is ready to merge. Extra tweaks should be proposed in additional PRs.

**Comment by philippjfr** [comment]
Looks fine to me. Happy to merge.

**Pull request merged**